src/configuration.rs

```diff
@@ -0,0 +1,588 @@
+            policy,
+            // TODO: SELinux could probably more-conveniently be represented as a HashMap
+            // TODO: last one wins for SELinux associates with the same name
+            selinux: match value.selinux {
+                Some(s) => s.associate,
+                None => vec![],
+            },
+            servicedir: value.servicedir,
+            servicehelper: value.servicehelper,
+            standard_session_servicedirs: value.standard_session_servicedirs.map(|_| true),
+            standard_system_servicedirs: value.standard_system_servicedirs.map(|_| true),
+            syslog: value.syslog.map(|_| true),
+            ..Default::default()
+        };
+
+        // > The last element "wins"
+        if let Some(te) = value.r#type.into_iter().last() {
+            bc.r#type = Some(te.text);
+        }
+        if let Some(ue) = value.user.into_iter().last() {
+            bc.user = Some(ue.text);
+        }
+
+        Ok(bc)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Error {
+    PolicyHasMultipleAttributes,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+enum IgnoreMissing {
+    No,
+    Yes,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+enum Include {
+    Session,
+    System,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+struct IncludeElement {
+    #[serde(rename = "@ignore_missing")]
+    ignore_missing: Option<IgnoreMissing>,
+    #[serde(rename = "$text")]
+    text: PathBuf,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+struct LimitElement {
+    #[serde(rename = "@name")]
+    name: LimitName,
+    #[serde(rename = "$text")]
+    text: i32, // semantically should be u32, but i32 for compatibility
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+enum LimitName {
+    AuthTimeout,
+    MaxCompletedConnections,
+    MaxConnectionsPerUser,
+    MaxIncomingBytes,
+    MaxIncomingUnixFds,
+    MaxIncompleteConnections,
+    MaxMatchRulesPerConnection,
+    MaxMessageSize,
+    MaxMessageUnixFds,
+    MaxNamesPerConnection,
+    MaxOutgoingBytes,
+    MaxOutgoingUnixFds,
+    MaxPendingServiceStarts,
+    MaxRepliesPerConnection,
+    PendingFdTimeout,
+    ServiceStartTimeout,
+    ReplyTimeout,
+}
+
+// reuse this between Vec<PathBuf> fields,
+// except those with field-specific attributes
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+struct PathBufElement {
+    #[serde(rename = "$text")]
+    text: PathBuf,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+enum Policy {
+    Console { rules: Vec<Rule> },
+    DefaultContext { rules: Vec<Rule> },
+    Group { group: Principal, rules: Vec<Rule> },
+    MandatoryContext { rules: Vec<Rule> },
+    NoConsole { rules: Vec<Rule> },
+    User { user: Principal, rules: Vec<Rule> },
+}
```

Comment:
I was particularly pleased with how this turned out, inspired by: https://dbus.freedesktop.org/doc/dbus-daemon.1.html

> The <policy> element has one of four attributes

I took the liberty of processing the RawPolicy into a handy Policy enum, so consumers don't have to check and validate each RawPolicy to determine which valid type of Policy it actually is